### DISCUSSION TITLE
Using vite with "vite_rails" gem

### DISCUSSION THREAD

**Discussion by thejprice** [nickjj/docker-rails-example#36] [Q&A]
Hi Nick, sorry to bug again...I've got vite up and running using [vite_rails](https://github.com/ElMassimo/vite_ruby) with your docker example, using [vite_rails_docker_example](https://github.com/ElMassimo/vite_rails_docker_example) as the template. I was hoping that you could check out my commit [here](https://github.com/thejprice/docker-rails-example-vite/commit/f841f86d12d30f994a0178da5255f0b7d1a60352) and see if I have set it up in a way that you would think is best. I feel like I hacked it together, but wanted to see if you had tips or better ideas on how to incorporate vite running as it's own service. Thanks!

**Comment by nickjj** [comment]
Hey,

I never used it but I would make `vite` its own separate service as mentioned in your other discussion. You could replace esbuild in https://github.com/nickjj/docker-rails-example/blob/main/docker-compose.override.yml.example (the js service) and swap it with vite. Then you wouldn't need a run script function to start vite, and I would also get rid of the Procfile since Docker Compose lets you start up multiple services.

`node_modules/` is already ignored in the `.gitignore` too btw.

**Reply by jon-price-snke** [reply]
Thanks Nick! Sorry for the late reply. I got everything up and running with vite and it works like a charm :)

**Reply by nickjj** [reply]
Nice, no problem.

If you were up for it, it would be super beneficial to others if you added the steps you took to get it to work as a new reply in this discussion.

**Reply by jon-price-snke** [reply]
For sure, when I get some time, I plan on putting together a simple branch showing how to implement vite. Thanks again!
